test/util/utils.test.js

```diff
@@ -0,0 +1,110 @@
+// Copyright IBM Corp. 2013,2016. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+var utils = require('../lib/utils');
+var assert = require('assert');
+
+describe('Utils', function() {
+  describe('uploadInChunks', function() {
+    it('should call process function with the chunked arrays', function() {
+      var largeArray = ['item1', 'item2', 'item3'];
+      var calls = [];
+
+      utils.uploadInChunks(largeArray, 1, processFunction, function() {
+        assert.deepEqual(calls, [['item1'], ['item2'], ['item3']]);
+      });
+
+      function processFunction(array, cb) {
```

Comment:
The usual convention for structuring unit tests is arrange-act-assert. I find your tests difficult to follow, because the `processFunction` ("act" phase) is defined after the assertions.

Could you please re-arrange all tests in this file to follow arrange-act-assert? For example:

```js
// option A
    it('should call process function with the chunked arrays', function() {
      var largeArray = ['item1', 'item2', 'item3'];
      var calls = [];

      function processFunction(array, cb) {
        calls.push(array);
        cb();
      }

      utils.uploadInChunks(largeArray, 1, processFunction, function() {
        assert.deepEqual(calls, [['item1'], ['item2'], ['item3']]);
      });
    });

// or even
    it('should call process function with the chunked arrays', function() {
      var largeArray = ['item1', 'item2', 'item3'];
      var calls = [];

      utils.uploadInChunks(
        largeArray, 1, 
        function processFunction(smallArray, cb) { 
          calls.push(smallArray);
          cb(); 
        },
        function onDone() {
          assert.deepEqual(calls, [['item1'], ['item2'], ['item3']]);
      });
    });
```

You should also check that `uploadInChunks` did not fail with an error. Also many (if not all) callback-based functions don't call the callback in the same tick of the event loop, therefore you should make all tests async:

```js
    it('should call process function with the chunked arrays', function(done) {
      var largeArray = ['item1', 'item2', 'item3'];
      var calls = [];

      utils.uploadInChunks(
        largeArray, 1, 
        function processFunction(smallArray, cb) { 
          calls.push(smallArray);
          cb(); 
        },
        function finished(err) {
          if (err) return done(err);
          assert.deepEqual(calls, [['item1'], ['item2'], ['item3']]);
          done();
      });
    });
```

The same comment applies to other new tests in this file too.